RevenueCatUI/Templates/V2/ViewHelpers/VisibleIfNeeded.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  VisibleIfNeeded.swift
+//
+//  Created by Josh Holtz on 2/13/25.
+
+import SwiftUI
+
+struct VisibleIfNeeded<Content: View>: View {
+
+    let visible: Bool
+    @ViewBuilder var content: () -> Content
+
+    var body: some View {
+        Group {
+            if visible {
+                self.content()
+            }
+        }
```

Comment:
Sorry for repeating myself @joshdholtz, but in case you missed my previous content. 
* The `group` is not needed here because `var body: some View` is a `@ViewBuilder`.
* Instead of introducing the `VisibleIfNeeded`, this could be easier solved adding the `@ViewBuilder` attribute to the `apply` parameters of the view models. E.g. 
```
@ViewBuilder
    func styles(
        state: ComponentViewState,
        condition: ScreenCondition,
        isEligibleForIntroOffer: Bool,
        @ViewBuilder apply: @escaping (IconComponentStyle) -> some View
```
I think that approach is cleaner and removes the need of the extra layer complexity layer of `VisibleIfNeeded`, which is what the `if` is for (in `@ViewBuilder`s)